### ISSUE TITLE
Refactor for upgradeable architecture with eternal storage

### ISSUE THREAD

**Issue by Ramarti** [EthicHub/platform-contracts#6]
Since we are going to store investors, localNodes, and other contract data in one contract, we may better refactor to a [hub and spoke architecture](https://medium.com/rocket-pool/upgradable-solidity-contract-design-54789205276d) 

This may allow us to upgrade Lending and Reputation in the future.

**Comment by kkx** [comment]
![acl](https://user-images.githubusercontent.com/1530467/39393907-27e7eb00-4acb-11e8-97cb-d8416fcf141c.png)
First version of an upgradable contracts structure with a separation of data and logic

**Comment by Ramarti** [comment]
After what we talked in telegram, we decided it's better to keep ACL with 1 responsibility: to determine if an address can set parameters in the storage. 
Making it also a façade (i.e. setReputation(address, uint)) adds unnecessary coupling between ACL and logic contracts, and adds one method call. 
We will use the generic get/sets.

The ACL will need to check if the contract is allowed to set to a specific key for better security. I suggest a parametric modifier, something like

`modifier allowedAddressForKey(bytes32 key)`
